src/gravel/api/nfs.py

```diff
@@ -138,9 +146,10 @@ async def export_create(
     '/export/{service_id}/{export_id}',
     name='delete an nfs export',
     response_model=Response)
-async def export_delete(service_id: str, export_id: int) -> Response:
+async def export_delete(request: Request, service_id: str, export_id: int) -> Response:
+    mgr = request.app.state.gstate.ceph_mgr
     try:
-        res = NFSExport().delete(service_id, export_id)
+        res = NFSExport(mgr).delete(service_id, export_id)
```

Comment:
(nit) a lot of boiler-plate with passing around the mon/mgr singletons. Any chance `__init__` can just pull them from gstate directly?

Reply:
Do you mean `__init__` on NFSExport?

Because we're instantiating here NFSExport doesn't have gstate. We could pass it instead of `mgr`, but we'd still need to pass in `request.app.state.gstate`.

This set of commits is designed to be followed through. I was thinking we could tidy things up to be more consistent with most classes just expecting `gstate` (and have `nodemgr` accessible from there etc). However I decided to leave that as a followup task to keep this from growing any longer (it's already 20 commits).